File: Unity Projekt/Assets/Scripts/CBR.Plan/BuildVillage.cs
using Assets.Scripts.Model;
using System.Runtime.Serialization;
namespace Assets.Scripts.CBR.Plan
{
    /**
     * Klasse, die die Aktion das Bauen von Siedlungen darstellt
     */
    [DataContract]
    public class BuildVillage : Action
    {
        /**
	     * Attribut zur Speicherung, in welcher Zeile gebaut werden soll 
	     */
        [DataMember]
        public int row;

        /**
         * Attribut zur Speicherung, in welcher Spalte gebaut werden soll 
         */
        [DataMember]
        public int column;


        public BuildVillage(int row, int column) : base("BuildVillage")
        {
            this.row = row;
            this.column = column;
        }

        public BuildVillage() : base("BuildVillage")
        {
            
        }
    }
}

File: Java Klassen/src/model/plan/BuildVillage.java
package model.plan;

import model.VillageBuildPlace;

/**
 * Klasse, die die Aktion das Bauen von Siedlungen darstellt
 * 
 * @author Tjark Harjes
 */
public class BuildVillage extends Action {

	private int row, column;

	public BuildVillage() {}

	
	public BuildVillage(VillageBuildPlace place) {
		this.setRow(place.row);
		this.setColumn(place.column);
	}


	public int getRow() {
		return row;
	}


	public void setRow(int row) {
		this.row = row;
	}


	public int getColumn() {
		return column;
	}


	public void setColumn(int column) {
		this.column = column;
	}
	
	
}
